### PR TITLE
My Jetpack: Add free search product support

### LINKED ISSUE #26767: My Jetpack: Update Search Purchase page to handle the pricing update
Location: /wp-admin/admin.php?page=my-jetpack#/add-search
Linked from: My Jetpack —  /wp-admin/admin.php?page=my-jetpack#/

This also resolves the issue of not being able to purchase Jetpack Search when only the Jetpack plugin is installed. Without the Search plugin, the Search Dashboard link on the wp-admin sidebar is not accessible.

### PR DESCRIPTION
**Pull request by kangzj**
Fixes #26767 

#### Changes proposed in this Pull Request:
- Add a status `needs_purchase_or_free` for product and set search to use it
- Add `wpcom_free_product_slug` for object `pricing_for_ui`
- Add `Start for free` button if `wpcom_free_product_slug` is not empty
- Adds `new_pricing_202208=1` in GET param to gate the feature

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
pcNPJE-1f2-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Apply this change to your Jetpack site.
* With a supported plugin installed (i.e. Jetpack or any of the standalone plugins), navigate to the My Jetpack page.
* Ensure that the label for Search has been updated to "Start for free" when `new_pricing_202208=1` added as a GET param.
* Ensure that the Search purchase page now features a new "Start for free" button below the "Add Jetpack Search" button when `new_pricing_202208=1` added as a GET param.

image

### CODE REVIEW

File: projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx [hunk 1]
```diff
@@ -211,7 +211,7 @@ export function SearchInterstitial() {
 			slug="search"
 			installsPlugin={ true }
 			supportingInfo={ __(
-				"Pricing will automatically adjust based on the number of records in your search index. If you grow into a new pricing tier, we'll let you know before your next billing cycle.",
+				"Jetpack Search Free supports up to 5,000 records and 500 search requests per month for free. Pricing will automatically adjust based on the number of records in your search index. If you grow into a new pricing tier, we'll let you know before your next billing cycle.",
```
Comment (jsnmoon) [inline projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx]: It'd be good to split this into two separate translatable strings.

File: projects/packages/my-jetpack/src/class-products.php [hunk 2]
```diff
@@ -152,7 +152,7 @@ public static function get_product_data_schema() {
 				'status'      => array(
 					'title' => 'The product status',
 					'type'  => 'string',
-					'enum'  => array( 'active', 'inactive', 'plugin_absent', 'needs_purchase', 'error' ),
+					'enum'  => array( 'active', 'inactive', 'plugin_absent', 'needs_purchase', 'needs_purchase_or_free', 'error' ),
```
Comment (jsnmoon) [inline projects/packages/my-jetpack/src/class-products.php:155]: Would it be possible to accomplish this without introducing `needs_purchase_or_free`?

**Review by jsnmoon — DISMISSED** [review]
Looks good overall.

This needs a changelogger bump and would also benefit from @Automattic/jetpack-agora's feedback.

**Review by jsnmoon — DISMISSED** [review]

**Comment by renatoagds** [comment]
I have a point about this one.

We're inserting a specific logic related with `search` inside the product card that, a component that should be agnostic to product logic, that's the reason we have an [specific card for each product](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/my-jetpack/_inc/components/product-cards-section), the [connected](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/my-jetpack/_inc/components/connected-product-card) one and the [generic](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/my-jetpack/_inc/components/product-card).

Looking at the feature, is just a matter of introducing a new status (`PRODUCT_STATUSES.TRIAL`: `trial` maybe ) and make sure we switch between `needs_purchase` and `trial` when passing down to `Product Card`.

Same feeling for `Product Detail Card` too.

**Comment by kangzj** [comment]
Thanks for the feedback @renatoagds 

> We're inserting a specific logic related with search inside components that should be agnostic to product logic

The special logic is only for gating the feature which will be removed when everything else is finished. I tried to make everything generic, so if another product have a similar situation, they could just add the free product slug and done ✅ 

But I am sure there are better ways. I don't quite get how we could apply your suggestions. Could you please add more details? 

Thanks 🙏

**Comment by jsnmoon** [comment]
> We're inserting a specific logic related with `search` inside the product card that, a component that should be agnostic to product logic, that's the reason we have an [specific card for each product](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/my-jetpack/_inc/components/product-cards-section), the [connected](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/my-jetpack/_inc/components/connected-product-card) one and the [generic](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/my-jetpack/_inc/components/product-card).
>
> [...]
>
> Same feeling for `Product Detail Card` too. (@renatoagds)

Hi Renato, I removed search-specific code from both ProductCard and ProductDetailCard components as you recommended. 

We don't have a strong opinion on how this needs to be implemented, but this does need to be merged and ready before we send out internal calls for testing next week. Might I suggest we go ahead with this PR and follow up in another PR, assuming we've resolved all the blockers?

**Comment by gibrown** [comment]
@renatoagds @jsnmoon @kangzj can we merge this yet?

**Comment by jsnmoon** [comment]
I just merged trunk into the feature branch. @gibrown: I'll merge this by my EOD, barring any blocking concerns.

**Review by dhasilva — DISMISSED** [review]

File: projects/packages/my-jetpack/_inc/components/product-card/index.jsx [hunk 3]
```diff
@@ -65,6 +67,12 @@ const ActionButton = ( {
 					{ __( 'Purchase', 'jetpack-my-jetpack' ) }
 				</Button>
 			);
+		case PRODUCT_STATUSES.NEEDS_PURCHASE_OR_FREE:
+			return (
+				<Button { ...buttonState } size="small" weight="regular" onClick={ onAdd }>
+					{ __( 'Try for free', 'jetpack-my-jetpack' ) }
```
Comment (kangzj) [inline projects/packages/my-jetpack/_inc/components/product-card/index.jsx]: Should we use `Start for free` as everywhere else?

Comment (jsnmoon) [reply projects/packages/my-jetpack/_inc/components/product-card/index.jsx]: Yes. I didn't realize that `Start for free` was our canonical phrase. I've reverted this change.

**Review by jsnmoon — DISMISSED** [review]

**Review by jsnmoon — APPROVED** [review]

**Pull request merged**